concordium-consensus/src/Concordium/Scheduler/EnvironmentImplementation.hs

```diff
@@ -240,98 +260,103 @@ instance
         return ret
 
     {-# INLINE removeBaker #-}
     removeBaker ai = do
-        s <- use schedulerBlockState
-        (ret, s') <- lift (bsoRemoveBaker s ai)
-        schedulerBlockState .= s'
+        s <- use ssBlockState
+        (ret, s') <- lift (BS.bsoRemoveBaker s ai)
+        ssBlockState .= s'
         return ret
 
     {-# INLINE updateBakerKeys #-}
     updateBakerKeys ai keyUpd = do
-        s <- use schedulerBlockState
-        (r, s') <- lift (bsoUpdateBakerKeys s ai keyUpd)
-        schedulerBlockState .= s'
+        s <- use ssBlockState
+        (r, s') <- lift (BS.bsoUpdateBakerKeys s ai keyUpd)
+        ssBlockState .= s'
         return r
 
     {-# INLINE updateBakerStake #-}
     updateBakerStake bi bsu = do
-        s <- use schedulerBlockState
-        (r, s') <- lift (bsoUpdateBakerStake s bi bsu)
-        schedulerBlockState .= s'
+        s <- use ssBlockState
+        (r, s') <- lift (BS.bsoUpdateBakerStake s bi bsu)
+        ssBlockState .= s'
         return r
 
     {-# INLINE updateBakerRestakeEarnings #-}
     updateBakerRestakeEarnings bi bre = do
-        s <- use schedulerBlockState
-        (r, s') <- lift (bsoUpdateBakerRestakeEarnings s bi bre)
-        schedulerBlockState .= s'
+        s <- use ssBlockState
+        (r, s') <- lift (BS.bsoUpdateBakerRestakeEarnings s bi bre)
+        ssBlockState .= s'
         return r
 
     {-# INLINE updateCredentialKeys #-}
     updateCredentialKeys accIndex credIndex newKeys = do
-        s <- use schedulerBlockState
-        s' <- lift (bsoSetAccountCredentialKeys s accIndex credIndex newKeys)
-        schedulerBlockState .= s'
+        s <- use ssBlockState
+        s' <- lift (BS.bsoSetAccountCredentialKeys s accIndex credIndex newKeys)
+        ssBlockState .= s'
 
     {-# INLINE getUpdateKeyCollection #-}
-    getUpdateKeyCollection = lift . bsoGetUpdateKeyCollection =<< use schedulerBlockState
+    getUpdateKeyCollection = lift . BS.bsoGetUpdateKeyCollection =<< use ssBlockState
 
     {-# INLINE getNextUpdateSequenceNumber #-}
     getNextUpdateSequenceNumber uty = do
-        s <- use schedulerBlockState
-        lift (bsoGetNextUpdateSequenceNumber s uty)
+        s <- use ssBlockState
+        lift (BS.bsoGetNextUpdateSequenceNumber s uty)
 
     {-# INLINE enqueueUpdate #-}
     enqueueUpdate tt p = do
-        s <- use schedulerBlockState
-        s' <- lift (bsoEnqueueUpdate s tt p)
-        schedulerBlockState .= s'
-
-deriving instance GS.BlockStateTypes (BSOMonadWrapper r state m)
-
-deriving instance AccountOperations m => AccountOperations (BSOMonadWrapper r state m)
-deriving instance ContractStateOperations m => ContractStateOperations (BSOMonadWrapper r state m)
-deriving instance ModuleQuery m => ModuleQuery (BSOMonadWrapper r state m)
+        s <- use ssBlockState
+        s' <- lift (BS.bsoEnqueueUpdate s tt p)
+        ssBlockState .= s'
+
+-- | Extract the underlying monad computation from the scheduler computation.
```

Comment:
I recommend that the documentation is more concrete telling the user what this is for. This comment is essentially meaningless unless you already know what it is doing from the type signature.

"Execute the computation using the provided context and initial scheduler state. The return value is the value produced by the computation and the updated state."